### PR TITLE
Add /list_projects_for_plan endpoint

### LINKED ISSUE #470: Backend API: get projects for a plan
We have `plan/create_project` and `plan/update_project`, and now we need `plan/get_projects` (returns a list of projects with display fields) and `plan/get_project` (returns a single project with all fields).

This will allow frontend to populate the "Drafted Scenarios" table so the user can resume planning.

image

### PR DESCRIPTION
**Pull request by elsiel23**
Returns a dictionary of Projects belonging to the owner with the given plan_id. May be empty if no Projects qualify.
Returns an error if the plan id does not exist in the database. 

Fixes #470

### CODE REVIEW

File: src/planscape/plan/views.py [hunk 1]
```diff
@@ -278,6 +278,20 @@ def update_project(request: HttpRequest) -> HttpResponse:
...
 
+@csrf_exempt
+def list_projects_for_plan(request: HttpRequest) -> HttpResponse:
+    try:
+        assert isinstance(request.GET['plan_id'], str)
+        plan_id = request.GET.get('plan_id', "0")
+
+        user = _get_user(request)
+
+        projects = Project.objects.filter(owner=user, plan=plan_id)
```
Comment (strangeties) [inline src/planscape/plan/views.py:289]: Should this say plan=int(plan_id)?

File: src/planscape/plan/urls.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 from django.urls import path
 from plan.views import (
     create_plan, create_project, delete, get_plan, get_project,
-    get_project_areas, get_scores, list_plans_by_owner, create_project_area, update_project)
+    get_project_areas, get_scores, list_plans_by_owner, create_project_area, update_project, list_projects_for_plan)
```
Comment (strangeties) [inline src/planscape/plan/urls.py:4]: nit: please add a line break.

File: src/planscape/plan/tests.py [hunk 3]
```diff
@@ -718,6 +718,50 @@ def test_get_project_with_priorities(self):
...
+
+    def test_list_no_plan_id(self):
+        self.client.force_login(self.user)
+        response = self.client.get(
+            reverse('plan:list_projects_for_plan'),
+            {},
+            content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_list_no_projects(self):
```
Comment (strangeties) [inline src/planscape/plan/tests.py:742]: nit: I can see this happening if a plan doesn't exist or if a plan has no projects. If a plan doesn't exist or if the user can't see the plan, would it be useful to return an error message instead?

**Review by thequeenofspades — APPROVED** [review]
thanks!!

File: src/planscape/plan/views.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:289]: I think the implicit conversion works, but I'll make it explicit

File: src/planscape/plan/tests.py (hunk 3 above)
Comment (elsiel23) [reply src/planscape/plan/tests.py:742]: test_list_no_plan_id somewhat covers (1) I can add a test for (2)

We return an error message with the error code and the message is generated by the django model framework i.e. something like Project for this id is not found. did you want to annotate the message?

Comment (strangeties) [reply src/planscape/plan/tests.py:742]: It'd be cool to see annotated error messages - but I also think it's more of a TODO than a blocker depending on FE needs.

**Review by strangeties — APPROVED** [review]

File: src/planscape/plan/tests.py (hunk 3 above)
Comment (elsiel23) [reply src/planscape/plan/tests.py]: sg!

**Pull request merged**